com.oracle.truffle.r.runtime/src/com/oracle/truffle/r/runtime/data/RAltIntVectorData.java

```diff
@@ -0,0 +1,286 @@
+            return ((RIntArrayVectorData) duplicatedObject).getIntDataCopy();
+        } else {
+            throw RInternalError.shouldNotReachHere(
+                    "RAltIntVectorData.getIntDataCopy: Unexpected returned object = " + duplicatedObject.toString()
+            );
+        }
+    }
+
+    @ExportMessage
+    public int[] getReadonlyIntData(@Cached AltrepRFFI.AltIntDataptrNode dataptrNode,
+                                    @Cached AltrepLengthNode lengthNode) {
+        return getDataCopy(dataptrNode, lengthNode);
+    }
+
+    private int[] getDataCopy(AltrepRFFI.AltIntDataptrNode dataptrNode, AltrepLengthNode lengthNode) {
+        long dataptrAddr = dataptrNode.execute(getOwner(), false);
+        int length = (int) lengthNode.execute(getOwner());
+        int[] dataCopy = new int[length];
+        NativeMemory.copyMemory(dataptrAddr, dataCopy, ElementType.INT, length);
+        return dataCopy;
+    }
+
+    @ExportMessage
+    public SeqIterator iterator(
+            @Shared("SeqItLoopProfile") @Cached("createCountingProfile()") LoopConditionProfile loopProfile,
+            @Cached AltrepLengthNode lengthnode) {
+        SeqIterator it = new SeqIterator(this, (int) lengthnode.execute(getOwner()));
+        it.initLoopConditionProfile(loopProfile);
+        return it;
+    }
+
+    @ExportMessage
+    public RandomAccessIterator randomAccessIterator() {
+        return new RandomAccessIterator(this);
+    }
+
+    @ExportMessage
+    public boolean next(SeqIterator it, boolean withWrap,
+                        @Shared("SeqItLoopProfile") @Cached("createCountingProfile()") LoopConditionProfile loopProfile) {
+        return it.next(loopProfile, withWrap);
+    }
+
+    @ExportMessage(limit = "1")
+    public int getIntAt(int index,
+                        @Cached(value = "create()") AltrepGetIntAtNode getIntAtNode) {
+        return (int) getIntAtNode.execute(getOwner(), index);
+    }
+
+    @ExportMessage(limit = "1")
+    public int getNextInt(SeqIterator it,
+                          @Cached(value = "create()") AltrepGetIntAtNode getIntAtNode) {
+        return (int) getIntAtNode.execute(getOwner(), it.getIndex());
+    }
+
+    @ExportMessage(limit = "1")
+    public int getInt(RandomAccessIterator it, int index,
+                      @Cached(value = "create()") AltrepGetIntAtNode getIntAtNode) {
+        return (int) getIntAtNode.execute(getOwner(), index);
+    }
+
+    // Write access to elements:
+
+    @ExportMessage
+    public SeqWriteIterator writeIterator(@Cached AltrepLengthNode lengthNode) {
+        int length = (int) lengthNode.execute(getOwner());
+        return new SeqWriteIterator(this, length);
+    }
+
+    @ExportMessage
+    public RandomAccessWriteIterator randomAccessWriteIterator() {
+        return new RandomAccessWriteIterator(this);
+    }
+
+    @ExportMessage
+    public void commitWriteIterator(SeqWriteIterator iterator, @SuppressWarnings("unused") boolean neverSeenNA) {
```

Comment:
If there is nothing to do in the `commit*` messages, you can just use the default implementation